high_card.rb

```diff
@@ -1,5 +1,57 @@
 # Use these two arrays to generate a deck of cards.
 ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
 suits = [ "hearts", "spades", "clubs", "diamonds" ]
-
+deck = []
 players = []
+cards_dealt = []
+
+
+# this makes the deck
+ranks.each do |rank|
+	suits.each do |suit|
+		deck << [rank, suit]
+	end
+end
+
+#shuffle the deck
+deck = deck.shuffle
+
+
+#create prompt
+loop do
+  puts "There are #{players.length} players so far. Enter a player name, or type 'play.'"
+	player = gets.chomp
+	players.push(:name => player)
+	break if player == "play"
+end
+
+
+
+#assign a card to each player. Store the card value in their object
+players.each do |player|
+	card = deck.pop
+	puts card
+	players.push(:dealt => card)
```

Comment:
If you leave this`players.push...` line, you're adding the dealt card as a new hash on the `players` array. Since we're inside of a loop on the `players` array, that means we never stop looping!

I think you meant `player` instead of `players`, and if so, `player` as a hash doesn't have a push method. Try this instead: `player[:deals] = card`